libcloud/compute/drivers/vsphere.py

```diff
@@ -49,506 +36,228 @@
-            if hasattr(e, 'strerror'):
-                message = e.strerror
-            fault = getattr(e, 'fault', None)
-
-            if fault == 'InvalidLoginFault':
-                raise InvalidCredsError(message)
-            raise LibcloudError(value=message, driver=self.driver)
-
-        atexit.register(self.disconnect)
-
-    def disconnect(self):
-        if not self.client:
-            return
-
-        try:
-            self.client.disconnect()
-        except Exception:
-            # Ignore all the disconnect errors
-            pass
-
-    def run_client_method(self, method_name, **method_kwargs):
-        method = getattr(self.client, method_name, None)
-        return method(**method_kwargs)
-
 
 class VSphereNodeDriver(NodeDriver):
     name = 'VMware vSphere'
     website = 'http://www.vmware.com/products/vsphere/'
     type = Provider.VSPHERE
-    connectionCls = VSphereConnection
 
     NODE_STATE_MAP = {
-        'POWERED ON': NodeState.RUNNING,
-        'POWERED OFF': NodeState.STOPPED,
-        'SUSPENDED': NodeState.SUSPENDED,
-        'POWERING ON': NodeState.PENDING,
-        'POWERING OFF': NodeState.PENDING,
-        'SUSPENDING': NodeState.PENDING,
-        'RESETTING': NodeState.PENDING,
-        'BLOCKED ON MSG': NodeState.ERROR,
-        'REVERTING TO SNAPSHOT': NodeState.PENDING
+        'poweredOn': NodeState.RUNNING,
+        'poweredOff': NodeState.STOPPED,
+        'suspended': NodeState.SUSPENDED,
     }
 
-    def __new__(cls, username, password, secure=True, host=None, port=None,
-                url=None, api_version=DEFAULT_API_VERSION, **kwargs):
-        if cls is VSphereNodeDriver:
-            if api_version == '5.5':
-                cls = VSphere_5_5_NodeDriver
+    def __init__(self, host, username, password):
+        """Initialize a connection by providing a hostname,
+        username and password
+        """
+        try:
+            self.connection = connect.SmartConnect(host=host, user=username,
+                                                   pwd=password)
+            atexit.register(connect.Disconnect, self.connection)
+        except Exception as exc:
+            error_message = str(exc).lower()
+            if 'incorrect user name' in error_message:
+                raise InvalidCredsError('Check your username and '
+                                        'password are valid')
+            if 'connection refused' in error_message or 'is not a vim server' \
+                                                        in error_message:
+                raise Exception('Check that the host provided is a '
+                                'vSphere installation')
+            if 'name or service not known' in error_message:
+                raise Exception('Check that the vSphere host is accessible')
+            if 'certificate verify failed' in error_message:
+                # bypass self signed certificates
+                try:
+                    import ssl
```

Comment:
if pyvmomi uses requests, then this probably won't work. It should expose `verify=` as a keyword argument.